function-processor/src/test/resources/wasm/Demo.java

```diff
@@ -0,0 +1,35 @@
+package chicory.testing;
+
+import com.dylibso.chicory.function.annotations.Allocate;
+import com.dylibso.chicory.function.annotations.Buffer;
+import com.dylibso.chicory.function.annotations.CString;
+import com.dylibso.chicory.function.annotations.Free;
+import com.dylibso.chicory.function.annotations.WasmExport;
+import com.dylibso.chicory.function.annotations.WasmModule;
+import com.dylibso.chicory.runtime.Instance;
+import com.dylibso.chicory.wasm.types.Value;
+
+@WasmModule
```

Comment:
This is more inline with what I was thinking. I like the annotation approach and providing explicit c types. What I think we're trying to accomplish is more akin to what JNA does today. Not necessarily write the high-level bindings for you yet, but give you some types and code to keep you from having to explicitly encode the ABI level details when we can just borrow from some known ABI and types (usually C). This reduces repetition and mistakes as @andreaTP pointed out. 

I can't make the call today but I'd vote for this approach. Feel free to make a decision without me though I trust y'all to figure out the right approach.